asf_heat_pump_suitability/analysis/exploratory/create_full_dataset/assign_cluster_suitability_and_feasibility.py

```diff
@@ -2,236 +2,402 @@
+    Prepares the dataframe for feasibility scoring. This includes:
+    - Transforming categorical variables into dummy/indicator variables.
+    - Creating new binary features based on existing columns and specified thresholds.
+    - Renaming columns for clarity.
 
     Args:
-        cluster_df (pl.DataFrame): DataFrame containing information about each cluster of properties.
+        df (pl.DataFrame): DataFrame containing the dataset with feature values for each UPRN.
+        features (list, optional): List of features used in the feasibility scoring.
+        anchor_dist_df (pl.DataFrame): DataFrame containing the distance (m) from the centre of each cluster to the nearest anchor load.
+        anchor_loads_threshold (float, optional): threshold for distance to anchor loads in metres.
+        outdoor_space_threshold (float, optional): Threshold for outdoor space in meters squared.
+            i.e. if garden_area_m2 > 0, then has_outdoor_space = True
+        city_centre_oas (set, optional): OAs that are considered part of the city center. Should be an oa21 format.
 
     Returns:
-        pl.DataFrame: cluster_df with an additional column for the suitability categorisation.
+        pl.DataFrame: DataFrame ready for feasibility scoring.
     """
 
-    cluster_df = cluster_df.with_columns(
-        pl.struct(["cluster_size", "hn_area", "city_centre", "has_outdoor_space"])
-        .map_elements(
-            lambda row: assign_str_suitable_techology_scheme(
-                row["cluster_size"],
-                row["hn_area"],
-                row["city_center"],
-                row["has_outdoor_space"],
+    # Transform predicted_tenure and predicted_property_type into dummies
+    df = df.to_dummies("predicted_tenure")
+    df = df.rename(
+        {
+            "predicted_tenure_owner-occupied": "owner_occupied",
+            "predicted_tenure_rental (social)": "social_housing",
+        }
+    )
+
+    df = df.to_dummies("predicted_property_type")
+    df = df.rename({"predicted_property_type_Flat, maisonette or apartment": "flats"})
+
+    # Compute on_gas
+    df = df.with_columns((~pl.col("use_off_gas")).alias("on_gas"))
+
+    # Compute not_in_listed_building
+    df = df.with_columns(
+        (~pl.col("in_listed_building")).alias("not_in_listed_building")
+    )
+    df = df.with_columns(
+        (~pl.col("in_conservation_area")).alias("not_in_conservation_area")
+    )
+
+    # Create has_outdoor_space from garden_area_m2
+    df = df.with_columns(
+        (pl.col("garden_area_m2") > outdoor_space_threshold).alias("has_outdoor_space")
+    )
+
+    # Creating close_to_city_centre from distance_to_city_centre and city_centre_threshold
+    df = df.with_columns(
+        pl.col("oa21").is_in(city_centre_oas).alias("close_to_city_centre")
+    )
+
+    # Creating imd_decile_above_avg from imd_decile
+    df = df.with_columns((pl.col("imd_decile") > 5).alias("imd_decile_above_avg"))
```

Comment:
```suggestion
    df = df.to_dummies("predicted_tenure").to_dummies("predicted_property_type")
        .rename(
            {
                "predicted_tenure_owner-occupied": "owner_occupied",
                "predicted_tenure_rental (social)": "social_housing",
                "predicted_property_type_Flat, maisonette or apartment": "flats"
            }
        ).with_columns(
                (~pl.col("use_off_gas")).alias("on_gas"),
                (~pl.col("in_listed_building")).alias("not_in_listed_building"),
                (~pl.col("in_conservation_area")).alias("not_in_conservation_area"),
                (pl.col("garden_area_m2") > outdoor_space_threshold).alias("has_outdoor_space"),
                pl.col("oa21").is_in(city_centre_oas).alias("close_to_city_centre"),
                (pl.col("imd_decile") > 5).alias("imd_decile_above_avg")
        )
```
`with_columns` allows you to add as many manipulations as you want into one chain :)